lib/bundler/build_metadata.rb

```diff
@@ -28,14 +28,19 @@ def self.git_commit_sha
       # If Bundler has been installed without its .git directory and without a
       # commit instance variable then we can't determine its commits SHA.
       git_dir = File.join(File.expand_path("../../..", __FILE__), ".git")
-      # Check for both a file or folder because RubyGems runs Bundler's test suite in a submodule
-      # which does not have a .git folder
-      return "unknown" unless File.exist?(git_dir)
+      if File.directory?(git_dir)
+        @git_commit_sha = Dir.chdir(git_dir) { `git rev-parse --short HEAD`.strip.freeze }
+      end
 
-      # Otherwise shell out to git.
-      @git_commit_sha = Dir.chdir(File.expand_path("..", __FILE__)) do
-        `git rev-parse --short HEAD`.strip.freeze
+      # If Bundler is a submodule in RubyGems, get the submodule commit
+      git_sub_dir = File.join(File.expand_path("../../../..", __FILE__), ".git")
+      if File.directory?(git_sub_dir)
+        @git_commit_sha = Dir.chdir(git_sub_dir) do
+          `git ls-tree --abbrev=8 HEAD bundler | awk '{ print $3 }'`.strip.freeze
```

Comment:
Maybe it's best to process output in ruby so we don't introduce a dependency on `awk`?

Reply:
good point! switched to `.split.fetch`